app/src/main/java/com/github/sdp_begreen/begreen/fragments/MapFragment.kt

```diff
@@ -1,70 +1,128 @@
 package com.github.sdp_begreen.begreen.fragments
 
+import android.Manifest
+import android.content.pm.PackageManager
+import android.location.Location
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
 import androidx.fragment.app.Fragment
 import com.github.sdp_begreen.begreen.R
 
-// TODO: Rename parameter arguments, choose names that match
-// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-private const val ARG_PARAM1 = "param1"
-private const val ARG_PARAM2 = "param2"
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
+
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+
 
 /**
  * A simple [Fragment] subclass.
  * Use the [MapFragment.newInstance] factory method to
  * create an instance of this fragment.
  */
 class MapFragment : Fragment() {
-    // TODO: Rename and change types of parameters
-    private var param1: String? = null
-    private var param2: String? = null
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        arguments?.let {
-            param1 = it.getString(ARG_PARAM1)
-            param2 = it.getString(ARG_PARAM2)
+
+    companion object {
+        private const val MAP_DEFAULT_ZOOM = 12f
+    }
+
+    private lateinit var map: GoogleMap
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private lateinit var lastLocation: Location
+
+    private val mapReadyCallback = OnMapReadyCallback { googleMap ->
+        /**
+         * Manipulates the map once available.
+         * This callback is triggered when the map is ready to be used.
+         * This is where we can add markers or lines, add listeners or move the camera.
+         */
+
+        map = googleMap
+        map.uiSettings.isZoomControlsEnabled = true
+
+        checkUserLocationPermissions()
+    }
+
+    private val requestPermissionLauncher = registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+
+        if (isGranted) {
+            displayUserLocation()
```

Comment:
Maybe you could add an `else` branch here to log that the user did not accept to share its location for now, and a `TODO` as a reminder to implement a mechanism to let know the user that as he refused, some functionality won't be available for him.

In order to be transparent with the user regarding his user experience.

So for now just add a `TODO`. What do you think about it ?

Reply:
I've added an else closure that displays a Toad _"Please grant the location access permissions in the settings"_.
In further updates, I could add a mechanism that allows the user to grant the permissions directly in the app (the system launches the permission window on a simple click).

As you said, I think that this feature is not mandatory right now, so we could discuss about it for further updates.

Reply:
I know you meant to write `Toast` but now I want an app that tells me to grant location access by showing this guy
![image](https://user-images.githubusercontent.com/859562/233843393-99931a58-49f2-47f8-963e-dda5250e896b.png)